forte/data/base_store.py

```diff
@@ -31,6 +32,81 @@ def __init__(self):
         various fields stored in entry lists.
         """
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def serialize(
+        self,
+        output_path: str,
+        serialize_method: str = "json",
+        save_attribute: bool = True,
+    ):
+        if serialize_method == "json":
+            with open(output_path, mode="wt", encoding="utf-8") as json_out:
+                json_out.write(self.to_string(serialize_method, save_attribute))
+        else:
+            raise NotImplementedError(
+                f"Unsupported serialization method {serialize_method}"
+            )
+
+    def to_string(
+        self,
+        json_method: str = "json",
+        save_attribute: bool = True,
+    ) -> str:
+        """
+        Return the string representation (json encoded) of this method.
+
+        Args:
+            json_method: What method is used to convert data pack to json.
+                Only supports `json` for now. Default value is `json`.
+            save_attribute: Boolean value indicating whether users want to
+                save `_type_attributes`.
+
+        Returns: String representation of the data pack.
+        """
+        if json_method == "json":
+            state = self.__getstate__()
+            if not save_attribute:
+                state.pop("fields")
+            return json.dumps(state, indent=2)
+        else:
+            raise ValueError(f"Unsupported JSON method {json_method}.")
+
+    @classmethod
+    def _deserialize(
+        cls,
+        data_source: str,
+        serialize_method: str = "json",
+    ):
+        """
+        This function should deserialize a data store from a string. The
+        implementation should decide the specific data store type.
+
+        Args:
+            data_source: The data path containing pack data. The content
+                of the data could be string or bytes depending on the method of
+                serialization.
+            serialize_method: The method used to serialize the data, this
+                should be the same as how serialization is done. The current
+                option is `json`。
+
+        Returns:
+            An data store object deserialized from the data.
+        """
+        if serialize_method == "json":
+            with open(data_source, mode="rt", encoding="utf8") as f:
+                pack = json.loads(f.read())
+            return pack
```

Comment:
how is `json.loads` giving us a `DataStore` object?

Reply:
it gives a dictionary. To recreate an object, I use `obj = DataStore()` and `obj.__setstate__(store)` in deserialize function in data_store.py